### PR TITLE
Fixes #784: Download specific version of rancher/agent

### PR DESCRIPTION
**Pull request by wlan0**
@will-chan @cjellick

### CODE REVIEW

**Comment by wlan0** [comment]
@cjellick I just noticed the Failed Build. Im looking into it.

File: handlers/bootstrap.go [hunk 1]
```diff
@@ -157,13 +155,13 @@ func pullImage(dockerClient *docker.Client) error {
 	return nil
 }
 
-var getRegistrationUrl = func(accountId string, apiClient *client.RancherClient) (string, error) {
+var getRegistrationUrl = func(accountId string, apiClient *client.RancherClient) (string, string, string, error) {
```
Comment (cjellick) [inline handlers/bootstrap.go]: Can you rename this to something like getUrlAndImage?

Comment (will-chan) [inline handlers/bootstrap.go]: This is probably picky but this function no longer just returns a URL.  You should rename it to something else like getRegistrationDetails or getRegistratioInfo.

Comment (will-chan) [inline handlers/bootstrap.go]: Craig was faster than me! I like his method name more.

**Comment by wlan0** [comment]
@will-chan and @cjellick -- named it getRegistrationUrlAndImage..

I created a droplet with the changes. Works fine. Haven't figure out the build failure yet.

**Comment by cjellick** [comment]
@wlan0 just to confirm, did you end-to-end test this by creating a machine in cattle and seeing that all the containers that were started on the machine (including the temporary/intermediate) ones were created with the pinned version of rancher/agent that is specified by cattle?

File: handlers/amazonec2_test.go [hunk 2]
```diff
@@ -67,8 +67,8 @@ func setup(accessKey string, secretKey string, vpcId string, zone string) {
 		return machine, nil
 	}
 
-	getRegistrationUrl = func(accountId string, apiClient *client.RancherClient) (string, error) {
-		return "http://1.2.3.4/v1", nil
+	getRegistrationUrlAndImage = func(accountId string, apiClient *client.RancherClient) (string, string, string, error) {
+		return "http://1.2.3.4/v1", "rancher/agent", "latest", nil
```
Comment (cjellick) [inline handlers/amazonec2_test.go]: Id rather see the test use a specific version. v0.7.6 is what is currently set in cattle, can you use that?

Comment (wlan0) [inline handlers/amazonec2_test.go]: ok

Comment (cjellick) [inline handlers/amazonec2_test.go]: @wlan0 are you going to change this?

**Comment by cjellick** [comment]
LGTM

**Pull request merged**